### ISSUE TITLE
FiniteSubgroupClasses runs into error inside ComplementClassesCR

### ISSUE THREAD

**Issue by fingolfin** [gap-packages/polycyclic#3]
```gap
gap> LoadPackage("crystcat",false);
true
gap> G := FiniteSubgroupClasses(PcGroupToPcpGroup(PcGroupQClass(4, 29, 9)));
#I  You are creating a Pc group with non-prime relative orders.
#I  Many algorithms require prime relative orders.
#I  Use `RefinedPcGroup' to convert.
#I  Warning: the presentation has been extended to get a prime order pcgs
Error, Assertion failure in
  Assert( 0, IsSubset( arg[1], pcs ) ); at /Users/mhorn/Projekte/GAP/repos/pkg/polycyclic.git-remotehg/gap/basic/pcpgrps.gi:129 called from
SubgroupByIgs( C.group, gens ) at /Users/mhorn/Projekte/GAP/repos/pkg/polycyclic.git-remotehg/gap/cohom/grpcom.gi:223 called from
ComplementClassesCR( C ) at /Users/mhorn/Projekte/GAP/repos/pkg/polycyclic.git-remotehg/gap/pcpgrp/torsion.gi:379 called from
SupplementClassesCR( C ) at /Users/mhorn/Projekte/GAP/repos/pkg/polycyclic.git-remotehg/gap/pcpgrp/torsion.gi:455 called from
FiniteSubgroupClassesBySeries( G, PcpsOfEfaSeries( G )
 ) at /Users/mhorn/Projekte/GAP/repos/pkg/polycyclic.git-remotehg/gap/pcpgrp/torsion.gi:489 called from
<function "unknown">( <arguments> )
 called from read-eval loop at *stdin*:2
you may 'return;'
brk>
```

**Comment by fingolfin** [comment]
Actually, without that `Assert` (which does not exist in the release version of polycyclic, just in my local development version), this is the error I get (with GAP 4.9 resp. master):
```
gap> LoadPackage("crystcat",false);
true
gap> G := FiniteSubgroupClasses(PcGroupToPcpGroup(PcGroupQClass(4, 29, 9)));
#I  Warning: the presentation has been extended to get a prime order pcgs
Error, no method found! For debugging hints type ?Recovery from NoMethodFound
Error, no 1st choice method found for `*' on 2 arguments
The 1st argument is 'fail' which might point to an earlier problem
 at /Users/mhorn/Projekte/GAP/gap.github/lib/methsel2.g:250 called from
PROD_SCL_LIST_DEFAULT( nonlist, list ) at /Users/mhorn/Projekte/GAP/gap.github/lib/list.gi:3281 called from
cc.CocToCBElement( cc, c ) * cc.trf at /Users/mhorn/Projekte/GAP/repos/pkg/pkg/pkg/polycyclic/gap/cohom/grpcom.gi:214 called from
ComplementClassesCR( D ) at /Users/mhorn/Projekte/GAP/repos/pkg/pkg/pkg/polycyclic/gap/pcpgrp/torsion.gi:391 called from
SupplementClassesCR( C ) at /Users/mhorn/Projekte/GAP/repos/pkg/pkg/pkg/polycyclic/gap/pcpgrp/torsion.gi:455 called from
FiniteSubgroupClassesBySeries( G, PcpsOfEfaSeries( G ) ) at /Users/mhorn/Projekte/GAP/repos/pkg/pkg/pkg/polycyclic/gap/pcpgrp/torsion.gi:489 called from
...  at *stdin*:2
you can 'quit;' to quit to outer loop, or
you can 'return;' to continue
brk> nonlist;
fail
brk> list;
[ Z(3)^0 ]
brk> cc.CocToCBElement(cc, c);
fail
brk> cc;
rec( CocToCBElement := function( coh, coc ) ... end, CocToCCElement := function( coh, coc ) ... end, CocToFactor := function( coh, coc ) ... end,
  ElementToCoc := function( gc, elm ) ... end, FactorToCoc := function( coh, elm ) ... end, action := [ 1 ],
  factor := rec( denom := [ [ Z(3)^0, 0*Z(3) ] ], gens := [  ], imgs := [ [  ] ], prei := [  ], rels := [  ] ), fld := GF(3), gcb := [ [ Z(3)^0, 0*Z(3) ] ],
  gcc := [ [ Z(3)^0, 0*Z(3) ] ], invgcc := [ [ Z(3)^0, 0*Z(3) ], [ 0*Z(3), Z(3)^0 ] ], rls := [  ], sol := [ 0*Z(3), Z(3)^0 ], trf := [ [ Z(3)^0 ] ] )
brk> c;
[ 0*Z(3), Z(3) ]
```

**Comment by fingolfin** [comment]
So this now looks as if it may be a duplicate of (or at least closely related to) issue #2.

**Comment by fingolfin** [comment]
Fixed by commit 36bdee8edc6871cb45f8d33fd9c5e1d581b9c9a7 (which like most recent fixes, is due to @beick ).

Unfortunately, issue #2 still seems to happen
